docs/content/latest/secure/tls-encryption/tls-authentication.md

```diff
@@ -0,0 +1,230 @@
+```
+
+{{< note title="Note" >}}
+`node.<ip>.crt/key` are the default values for the `ssl_cert_file` and `ssl_key_file` server-side configuration for a YSQL node. If your local IP is not 127.0.0.1, then use the appropriate local IP to name the two files. Alternatively use `ysql_pg_conf` to set `ssl_cert_file` and `ssl_key_file` to the appropriate values.
+{{< /note >}}
+
+### TLS without authentication
+
+This configuration requires the client to use client-to-server encryption to connect.
+
+Create the database:
+
+```sh
+$ ./bin/yb-ctl destroy && ./bin/yb-ctl create --tserver_flags="$ENABLE_TLS"
+```
+
+To connect, SSL must be enabled in the client.
+
+```sh
+$ ./bin/ysqlsh "sslmode=disable"
+```
+
+```output
+ysqlsh: FATAL:  no pg_hba.conf entry for host "127.0.0.1", user "yugabyte", database "yugabyte", SSL off
+```
+
+```sh
+$ ./bin/ysqlsh "sslmode=require"
+```
+
+```output
+ysqlsh (11.2-YB-2.7.0.0-b0)
+SSL connection (protocol: TLSv1.2, cipher: ECDHE-RSA-AES256-GCM-SHA384, bits: 256, compression: off)
+Type "help" for help.
+```
+
+The default `ysqlsh` SSL mode is `prefer` (see <https://www.postgresql.org/docs/11/libpq-ssl.html>), which tries SSL first, but falls back to no-ssl if the server does not support it.
+
+In this case, a plain `ysqlsh` with no options will work and use encryption:
+
+```sh
+$ ./bin/ysqlsh
+```
+
+```output
+ysqlsh (11.2-YB-2.7.0.0-b0)
+SSL connection (protocol: TLSv1.2, cipher: ECDHE-RSA-AES256-GCM-SHA384, bits: 256, compression: off)
+Type "help" for help.
+```
+
+{{< note title="Note" >}}
+The following examples omit setting `sslmode` for brevity.
+{{< /note >}}
+
+### TLS with authentication
+
+This configuration requires the client to use client-to-server encryption and authenticate with a password to connect.
+
+Create the database:
+
+```sh
+$ ./bin/yb-ctl destroy && ./bin/yb-ctl create --tserver_flags="$ENABLE_TLS,ysql_enable_auth=true"
+```
+
+To connect to the database, the password is required (see second line below):
+
+```sh
+$ ./bin/ysqlsh
+```
+
+```output
+Password for user yugabyte:
+ysqlsh (11.2-YB-2.7.0.0-b0)
+SSL connection (protocol: TLSv1.2, cipher: ECDHE-RSA-AES256-GCM-SHA384, bits: 256, compression: off)
+Type "help" for help.
+```
+
+The other modes (that is, `sslmode=require` or `disable`) behave analogously.
+
+### TLS with authentication via certificate
+
+This configuration requires the client to use client-to-server encryption and authenticate with the appropriate certificate to connect.
+
+{{< note title="Note" >}}
+This was the previous default for TLS without authentication; this example shows the `ysql_hba_conf_csv` configuration to use to replicate the previous behavior.
+{{< /note >}}
+
+Create the database:
+
+```sh
+$ ./bin/yb-ctl destroy && ./bin/yb-ctl create \
+    --tserver_flags="$ENABLE_TLS" \
+    --ysql_hba_conf_csv="hostssl all all all cert"
+```
+
+To connect, a certificate is required, but no password.
```

Comment:
It would be good to add more explanations between commands and outputs.
Also the same comment about prompts